### PR TITLE
docs: sync the core features in README to index.rst

### PR DESCRIPTION
**Pull request by zhyncs**
## Motivation

as titled cc @lvhan028 @RunningLeon 
fix https://github.com/InternLM/lmdeploy/issues/1975 ref https://github.com/InternLM/lmdeploy/issues/1975#issuecomment-2220157740

## Modification

as titled

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by lvhan028** [comment]
image

**Comment by lvhan028** [comment]
@zhyncs Pls use the readthedocs url as the hyperlinks  for [KV Cache Quant], [AWQ] and [Automatic Prefix Caching]

File: docs/en/index.rst [hunk 1]
```diff
@@ -31,6 +31,7 @@ LMDeploy has the following core features:
 
 * **Interactive Inference Mode**: By caching the k/v of attention during multi-round dialogue processes, the engine remembers dialogue history, thus avoiding repetitive processing of historical sessions.
 
+* **Excellent Compatibility**: LMDeploy supports [KV Cache Quant](docs/en/quantization/kv_quant.md), [AWQ](docs/en/quantization/w4a16.md) and [Automatic Prefix Caching](docs/en/inference/turbomind_config.md) to be used simultaneously.
```
Comment (RunningLeon) [inline docs/en/index.rst:34]:
```text
`KV Cache Quant <https://lmdeploy.readthedocs.io/en/latest/quantization/kv_quant.html>`_, 
`AWQ <https://lmdeploy.readthedocs.io/en/latest/quantization/w4a16.html>`_ 
`Automatic Prefix Caching <https://lmdeploy.readthedocs.io/en/latest/quantization/w4a16.html>`_
```

**Comment by zhyncs** [comment]
> @zhyncs Pls use the readthedocs url as the hyperlinks for [KV Cache Quant], [AWQ] and [Automatic Prefix Caching]

ok

File: docs/en/index.rst (hunk 1 above)
Comment (zhyncs) [reply docs/en/index.rst:34]: ok

Comment (RunningLeon) [reply docs/en/index.rst]: not rendered correctly as you see here
https://lmdeploy--1988.org.readthedocs.build/en/1988/

Comment (zhyncs) [reply docs/en/index.rst]: ok

Comment (zhyncs) [reply docs/en/index.rst]: It works now. Thanks! https://lmdeploy--1988.org.readthedocs.build/en/1988/

File: docs/zh_cn/index.rst [hunk 2]
```diff
@@ -31,6 +31,7 @@ LMDeploy 工具箱提供以下核心功能：
 
 - **有状态推理：** 通过缓存多轮对话过程中 attention 的 k/v，记住对话历史，从而避免重复处理历史会话。显著提升长文本多轮对话场景中的效率。
 
+- **卓越的兼容性:**  LMDeploy 支持 `KV Cache 量化 <https://lmdeploy.readthedocs.io/en/latest/quantization/kv_quant.html>`_, `AWQ <https://lmdeploy.readthedocs.io/en/latest/quantization/w4a16.html>`_ 和 `Automatic Prefix Caching <https://lmdeploy.readthedocs.io/en/latest/inference/turbomind_config.html>`_ 同时使用。
```
Comment (lvhan028) [inline docs/zh_cn/index.rst:34]: Let's use the zh_cn links

Comment (zhyncs) [reply docs/zh_cn/index.rst:34]: ok

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**